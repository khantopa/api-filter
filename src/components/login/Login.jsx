import React, { Component } from 'react';
import User from'../../images/user-white.svg';
import Email from '../../images/email.svg';
import Key from '../../images/key.svg'
import './Login.scss';


export default class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      status: '',
      isLogedIn: false
    }
  }

  handleChange = (e) => {
    let name = e.target.name;
    let value = e.target.value
    this.setState({
      [name]:value,
      status: ''
    })
  }

  login = () => {
    const email = this.state.email;
    const password = this.state.password;
    if(email === "test@test.com" && password === "12345678") {
      this.props.history.push('/contacts');
    } else {
      this.setState({ status: 'Incorrect username and password'})
    }
  }

  render() {
    return (
      <div className="login">
        <div className="login_card">
          <div className="login_icon">
            <img className="login_icon_img" src={User} alt="User-icon" />
          </div>
          <div className="login_form">
            <label>
              <input type="email" className="login_form_field" placeholder="E-mail" 
              name="email" onChange={ this.handleChange } id="email" /> 
              <img className="icon" src={ Email } alt="Email" /> 
            </label>
            <label>
              <input type="password" className="login_form_field" placeholder="Password" 
              name="password" onChange={ this.handleChange } id="password" />
              <img className="icon" src={ Key } alt="Key" />
            </label>
          </div>
          { this.state.status !== ''
            ? <div className="error" style={{'background': 'red'}}>
                <p>{ this.state.status }</p>
              </div>
            : <div className="error"></div>
          }
          <br/><br/>
          <div className="login_submit">
            <button onClick={ this.login } className="login_submit_button">Login</button>
          </div>
        </div>
      </div>
    )
  }
}
