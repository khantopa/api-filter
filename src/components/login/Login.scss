@import '../../styles/main.scss';

.login {
  position: relative;
  width: 100vw;
  height: 100vh;
  overflow: hidden;
  background: url('../../images/bg.jpg') center/cover no-repeat;
  color: $light;

  &_icon {
    @include grid(1);
    place-items: center;
    &_img {
      height: 80px;
      width: 80px;
    }
  }
  &_card {
    @include absPosition(50%, 50%);
    @include card(320px, auto);
    transform: translate(-50%, -50%);
  }

  &_form {
    padding: 30px 0 10px 0;
    @include grid(1);
    grid-gap: 20px;
    &_field {
      @extend .form_field;
      text-align: left;
      padding-left: 40px;
    }
    label {
      position: relative;
      .icon {        
        @include absPosition(4%, 50%);
        transform: translateY(-50%);
        width: 20px;
      }
    }
  }
  &_submit {
    @include grid(1);
    &_button {
      @extend .button;
      width: 40%;
      background: $dark;
    }
  }
} 