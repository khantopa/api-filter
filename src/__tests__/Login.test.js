import Login from '../components/login/Login';

let wrapper;

beforeEach(() => {
  wrapper = shallow(<Login history={[]} />);
});

afterEach(() => {
  wrapper.unmount();
});

it('should render login', () => {
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it('Email check', () => {
  wrapper.find('input[type="email"]').simulate('change', { target: { name: 'email', value: 'test@test.com' } });
  expect(wrapper.state('email')).toEqual('test@test.com');
});

it('Password check', () => {
  wrapper.find('input[type="password"]').simulate('change', { target: { name: 'password', value: '12345678' } });
  expect(wrapper.state('password')).toEqual('12345678');
});

it('login check with right data', () => {
  wrapper.find('input#email').simulate('change', { target: { name: 'email', value: 'test@test.com' } });
  wrapper.find('input#password').simulate('change', { target: { name: 'password', value: '12345678' } });
  wrapper.find('button').simulate('click'); 
  expect(wrapper.state('email')).toEqual('test@test.com');
  expect(wrapper.state('password')).toEqual('12345678');
})