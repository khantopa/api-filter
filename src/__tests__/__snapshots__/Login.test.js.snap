// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render login 1`] = `
<div
  className="login"
>
  <div
    className="login_card"
  >
    <div
      className="login_icon"
    >
      <img
        alt="User-icon"
        className="login_icon_img"
        src="user-white.svg"
      />
    </div>
    <div
      className="login_form"
    >
      <label>
        <input
          className="login_form_field"
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="E-mail"
          type="email"
        />
        <img
          alt="Email"
          className="icon"
          src="email.svg"
        />
      </label>
      <label>
        <input
          className="login_form_field"
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
        />
        <img
          alt="Key"
          className="icon"
          src="key.svg"
        />
      </label>
    </div>
    <div
      className="error"
    />
    <br />
    <br />
    <div
      className="login_submit"
    >
      <button
        className="login_submit_button"
        onClick={[Function]}
      >
        Login
      </button>
    </div>
  </div>
</div>
`;
